Adding Centos and Amazon Linux support

jpooler opened the pull request:
### Description

[Describe what this change achieves]

### Issues Resolved

[List any existing issues this PR resolves]

### Check List

- [ ] All tests pass. See <https://github.com/chef-cookbooks/community_cookbook_documentation/blob/master/TESTING.MD>
- [ ] New functionality includes testing.
- [ ] New functionality has been documented in the README if applicable
- [ ] All commits have been signed for the Developer Certificate of Origin. See <https://github.com/chef-cookbooks/community_cookbook_documentation/blob/master/CONTRIBUTING.MD>

tas50 commented on resources/install.rb:
```diff
@@ -57,7 +57,11 @@ def db_package
+      if node['platform'] == 'amazon'
+        'db4-utils'
+      else
+        node['platform_version'].to_i >= 7 ? 'compat-db47' : 'db4-utils'
```
node['platform_version'].to_i >= 7 && !platform?('amazon') ? 'compat-db47' : 'db4-utils' would accomplish this without the if/else

tas50 commented on recipes/default.rb:
```diff
@@ -27,8 +27,14 @@
 when 'rhel'
-  template '/etc/sysconfig/slapd' do
-    source 'sysconfig_slapd.erb'
+  if node['platform'] == 'amazon' || node['platform'] == 'centos'
```
You can use this here to simplify things:

```ruby
if platform?('amazon','centos')
```

cheeseplus commented:
I think I figured out why centos-6 is failing - it appears if you set the SLAPD_URLS when the defaults for SLAPD_LDAP et al are true it tries to listen twice on the same interface and fails out.

validated using below changes to sysconfig_slapd.rb => /etc/openldap/ldap
```
# Options of slapd (see man slapd)
<% if node['platform'] == "centos" || node['platform'] == "amazon" -%>
SLAPD_OPTIONS="-f /etc/openldap/slapd.conf"
<% else %>
SLAPD_OPTIONS=""
<% end -%>

# At least one of SLAPD_LDAP, SLAPD_LDAPI and SLAPD_LDAPS must be set to 'yes'!
#
# Run slapd with -h "... ldap:/// ..."
#   yes/no, default: yes
SLAPD_LDAP=yes

# Run slapd with -h "... ldapi:/// ..."
#   yes/no, default: yes
SLAPD_LDAPI=yes

# Run slapd with -h "... ldaps:/// ..."
#   yes/no, default: no
<% if node['openldap']['ldaps_enabled'] -%>
SLAPD_LDAPS=yes
<% else -%>
SLAPD_LDAPS=no
<% end -%>

# Run slapd with -h "... $SLAPD_URLS ..."
# This option could be used instead of previous three ones, but:
# - it doesn't overwrite settings of $SLAPD_LDAP, $SLAPD_LDAPS and $SLAPD_LDAPI options
# - it isn't overwritten by settings of $SLAPD_LDAP, $SLAPD_LDAPS and $SLAPD_LDAPI options
# example: SLAPD_URLS="ldapi:///var/lib/ldap_root/ldapi ldapi:/// ldaps:///"
# default: empty
#SLAPD_URLS=""

# Maximum allowed time to wait for slapd shutdown on 'service ldap stop' (in seconds)
#SLAPD_SHUTDOWN_TIMEOUT=3

# Parameters to ulimit, use to change system limits for slapd
#SLAPD_ULIMIT_SETTINGS=""
```

jpooler commented:
Rebased to squash commits and fixed the cookstyle issue (TIL cookstyle). I'm confused why it's getting the SLAPD_URLS since i'm not enabling TLS (yet). I'm guessing it's in the testing suite but haven't dug further into it. Next steps? I ran kitchen converge default-centos-68 locally and it succeeded. I'll be in #chef the rest of the day if you want to ping.

cheeseplus commented:
It gets the SLAPD_URLS because that is how we're setting it across the board, there are subtle differences between using SLAPD_LDAP, SLAPD_LDAPS, and SLAPD_LDAPI vs SLAPD_URLS which seem to exclusively apply to the CentOS 6 startup scripts.

cheeseplus commented:
Also for posterity it seems that the sysconfig file name is different on CentOS/RHEL 7 `/etc/sysconfig/ldap` => `/etc/sysconfig/slapd` so the logic will need to be fixed one way or another.

cheeseplus commented:
Going to merge this as is and then fix up the other issue this uncovered in a following PR.
